### PR TITLE
use MVCCChain in delete

### PR DESCRIPTION
**Pull request by jiangxinmeng1**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [x] Code Refactoring

## Which issue(s) this PR fixes:

issue #4828 

## What this PR does / why we need it:
1. Use MVCCChain and TxnMVCCNode in deletenode and deletechain

### CODE REVIEW

File: pkg/vm/engine/tae/catalog/basemvccnode.go [hunk 1]
```diff
@@ -91,7 +91,7 @@ func (un *EntryMVCCNode) PrepareCommit() (err error) {
 	return nil
 }
 func (un *EntryMVCCNode) String() string {
-	return fmt.Sprintf("CreatedAt=%v,DeletedAt=%v", un.CreatedAt, un.DeletedAt)
+	return fmt.Sprintf("CreatedAt=%s,DeletedAt=%s", un.CreatedAt.ToString(), un.DeletedAt.ToString())
```
Comment (XuPeng-SH) [inline pkg/vm/engine/tae/catalog/basemvccnode.go:94]: CreateAt=%s,DeleteAt=%s ---->  C@%s,D@%s

File: pkg/vm/engine/tae/catalog/segment.go [hunk 2]
```diff
@@ -163,6 +163,20 @@ func (entry *SegmentEntry) String() string {
...
 
+func (entry *SegmentEntry) StringWithLevel(level common.PPLevel) string {
+	entry.RLock()
+	defer entry.RUnlock()
+	return entry.StringWithLevelLocked(level)
+}
+
+func (entry *SegmentEntry) StringWithLevelLocked(level common.PPLevel) string {
+	if level <= common.PPL1 {
+		return fmt.Sprintf("[%s]SEGMENT[%d][CreateAt=%s,DeleteAt=%s]",
```
Comment (XuPeng-SH) [inline pkg/vm/engine/tae/catalog/segment.go:174]: TABLE --> TBL
SEGMENT --> SEG
BLOCK --> BLK
CreatedAt --> C@
DeletedAt --> D@
LogIndex --> LIDX
APPENDABLE --> A
NONAPPENDABLE --> NA

**Review by XuPeng-SH — CHANGES_REQUESTED** [review]

**Review by XuPeng-SH — APPROVED** [review]
lgtm

**Comment by jiangxinmeng1** [comment]
@mergifyio refresh

**Pull request merged**